### ISSUE TITLE
Log an additional event every time a duplicate repo is added

### ISSUE THREAD

**Issue by jplahn** [sourcegraph/sourcegraph#27795]
Details in this slack thread: https://sourcegraph.slack.com/archives/CHPC7UX16/p1626822437340100

TL;DR; the table (public.repo) with data on all the repos added to cloud (including the date they were added) does not contain any duplicates (details on tables in this doc). So if User A adds Repo1 on 2021-07-15 and User B also adds Repo1, but three days later on 2021-07-18, Repo1 shows up once in the public.repo table, with a created date of 2021-07-15.

**What this means:**
* Continuing the above scenario: the data will show that User A and User B both added Repo1 to their sourcegraph account on 2021-07-15
* TL;DR any “repos added/deleted over time” analysis will be inaccurate using existing data, for any repo that was added more than once

**Comment by ryanslade** [comment]
We have the `external_service_repos` table that records the mapping between a repo and the external service that added it. It also included either the `user_id` or `org_id` depending on who owns the external service. All users add repos via their own external service on Cloud so we could add a `created_at` column to this table to track when the repo was added by a specific user.
